Core/Src/main.c

```diff
@@ -87,17 +91,37 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
-
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);  // Start PWM on channel 1
+  uint8_t rxData[3];
+  uint8_t txData[3];
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	// Set the chip select low to start communication
+	HAL_GPIO_WritePin(GPIOA, GPIO_PIN_8, GPIO_PIN_RESET);
+
+	txData[0] = 0x1;  // 00000001 for start bit
+	txData[1] = 0x80; // 10000000 for single-ended on CH1
+
+	HAL_SPI_TransmitReceive(&hspi1, txData, rxData, 3, HAL_MAX_DELAY);
+	uint16_t adc_value = ((rxData[1] << 8) & (0x3 << 8)) | rxData[2];
```

Comment:
Let's make it simpler to see - (rxData[1] & 0x03) << 8.

Reply:
https://github.com/UWARG/embedded-bootcamp/pull/93/commits/680784f506c7680f26dadad51d71b19d85790091